controllers/clusterurlmonitor/clusterurlmonitor_supplement.go

```diff
@@ -0,0 +1,99 @@
+package clusterurlmonitor
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/route-monitor-operator/api/v1alpha1"
+	"github.com/openshift/route-monitor-operator/pkg/blackboxexporter"
+	"github.com/openshift/route-monitor-operator/pkg/const/blackbox"
+	utilfinalizer "github.com/openshift/route-monitor-operator/pkg/util/finalizer"
+	"github.com/openshift/route-monitor-operator/pkg/util/reconcile"
+	"github.com/openshift/route-monitor-operator/pkg/util/templates"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ClusterUrlMonitorSupplement struct {
+	clusterUrlMonitor v1alpha1.ClusterUrlMonitor
+	client            client.Client
+	log               logr.Logger
+	ctx               context.Context
+	blackboxExporter  *blackboxexporter.BlackboxExporter
+}
+
+func NewSupplement(clusterUrlMonitor v1alpha1.ClusterUrlMonitor, client client.Client, log logr.Logger, blackboxExporter *blackboxexporter.BlackboxExporter) *ClusterUrlMonitorSupplement {
+	return &ClusterUrlMonitorSupplement{clusterUrlMonitor, client, log, context.Background(), blackboxExporter}
+}
+
+func (s *ClusterUrlMonitorSupplement) EnsureFinalizer() (reconcile.Result, error) {
```

Comment:
this is also common logic (besides the finalizer key)

should this be refactored

Reply:
Not sure if it's worth the effort given it'll need to be agnostic of the type. In the end I think I would wrap that common functionality again in a function here.